### ISSUE TITLE
the single_frame in mydata

### ISSUE THREAD

**Issue by zj19921221** [clks-wzz/FAS-SGTD#7]
hi thanks for you release the code;
I have run the single frame code on  my  own data of face_anti_spoofing;
in my opnion, the key point of  the single frame method is the  "Depthwise Spatial
Gradient Magnitude" but I found out that  in my exprienment,  it is not play a good role in my dataset;
 how about your dataset; 
on more question : Was  the  Depthwise Spatial Gradient Magnitude better than "short cut" in you expriment? 

looking forward for your reply! 
thanks

**Comment by clks-wzz** [comment]
What's the setting in your experiments, such as backbone, learning rate and training iteration?

**Comment by zj19921221** [comment]
thanks for your reply;
1、Have you compared Depthwise Spatial Gradient Magnitude with short_cut which keeped all parameters  same;
2、in my experiment, backbone is mobilenet_v2 ; I replace the short_cut with the "Depthwise Spatial Gradient Magnitude" and keeped the parameters all same, I found out that "Depthwise Spatial Gradient Magnitude" is not as well as in my exp and data;  how about you?

**Comment by clks-wzz** [comment]
The mobilenet_v2 may be too deep to converge for RSGB and depth supervised learning. Which loss function do you try in your experiment, binary or depth?  A pretrained model initialization on IMAGENET might bring you a better result.

**Comment by zj19921221** [comment]
my loss: binary and depth;  
thanks for your reply!
I have another question why you use architechture in the single_frame_net;
DO you use other models such as mobilenet_v2 shuffenet_v2 ?
you Net is better than those models?

thanks

**Comment by zj19921221** [comment]
> The mobilenet_v2 may be too deep to converge for RSGB and depth supervised learning. Which loss function do you try in your experiment, binary or depth? A pretrained model initialization on IMAGENET might bring you a better result.

hi  wzz
you mentioned that mobilenet_v2 is too deep;  RSGB is more suitable used in the former layers?

**Comment by clks-wzz** [comment]
We have tried deeper network architectures, and found that the deeper models even may be not suitable to the depth supervision task. 
You can try RSGB in the former layers, or directly use shallower network architectures.

**Comment by zj19921221** [comment]
ok thanks for your reply

**Comment by shahrzadesmat** [comment]
I want to convert all of the files(modules) to one .ipynb file and I have problems and errors could you please help me with sending me the correct code...I am working on it for one of my master courses!

**Comment by shahrzadesmat** [comment]
> hi thanks for you release the code;
> I have run the single frame code on my own data of face_anti_spoofing;
> in my opnion, the key point of the single frame method is the "Depthwise Spatial
> Gradient Magnitude" but I found out that in my exprienment, it is not play a good role in my dataset;
> how about your dataset;
> on more question : Was the Depthwise Spatial Gradient Magnitude better than "short cut" in you expriment?
> 
> looking forward for your reply!
> thanks

could you please share your code and your data?thanksss

**Comment by punitha-valli** [comment]
@clks-wzz @zj19921221 @shahrzadesmat 

Hi,

I would like to know about the test score, in the util_test_OULU_Protocol_1.py

the Testscor.txt, file

can you please tell me about that text file? how to generate those text files.

How did you calculating the map score?

it will be a great help for my studies

hope for a reply

thank you
